jetstream/maxtext/http-server/http_server.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""HTTP Server to interact with MaxText + JetStream Server."""
+
+import asyncio
+import concurrent.futures
+import json
+import logging
+from typing import Optional
+
+import fastapi
+import grpc
+from jetstream.core.proto import jetstream_pb2
+from jetstream.core.proto import jetstream_pb2_grpc
+import pydantic
+
+
+class GenerateRequest(pydantic.BaseModel):
+  server: Optional[str] = "127.0.0.1"
+  port: Optional[str] = "9000"
+  session_cache: Optional[str] = ""
+  prompt: Optional[str] = "AB"
+  priority: Optional[int] = 0
+  max_tokens: Optional[int] = 3
+
+
+app = fastapi.FastAPI()
+executor = concurrent.futures.ThreadPoolExecutor(max_workers=200)
```

Comment:
How did we decide on 200 for the max workers? Also, how does it do under benchmarking when qps is over 200?

Reply:
The implementation of the http server again is not for users' reference, it's highly unlikely in a real case, this will be used since jetsream already creates a grpc server

Reply:
our benchmarking results is similar to CMCS's one, but what would be a reasonable value here though?

Reply:
If we benchmarked the http server in our benchmarking and it worked well enough for our max throughput case, then I'm ok with this value. In general, I want to make sure we choose this value with the understanding of what the limitations are and if it works for the majority of the use cases.

Reply:
Thanks for bringing this up Ashok. We're only done initial benchmarks right now so it's hard to determine. With Saxml, we found success with 1000. I'll change it here for now since I anticipate that the benchmarking will be similar between Saxml and Jetstream/MaxText. We can adjust the value further in the future as well as we see fit.

Reply:
thanks for bring this up, yes, let's use 1000 just in case there is any issue.